### PR TITLE
feat(invoice registry) optimize sql query

### PR DESCRIPTION
**Pull request by jeremielodi**
Joining invoice with debtor makes the request very slow, this PR solves that issue.

### CODE REVIEW

File: server/controllers/finance/patientInvoice.js [hunk 1]
```diff
@@ -227,6 +227,12 @@ function find(options) {
   // @FIXME Remove this with client side filter design
   delete options.patientNames;
 
+  let debtorJoin = ``;
+  const debtorKey = 'debtor_group_uuid';
```
Comment (jniles) [inline server/controllers/finance/patientInvoice.js:231]: Are you sure this shouldn't be `debtor_uuid`?  Why filter on the `debtor_group_uuid`?

Comment (jeremielodi) [reply server/controllers/finance/patientInvoice.js:231]: I have not changed the join query to debtor. The only thing I tried to do is to add a flag that test if the user sent ```debtor_group_uuid ``` in ```req.query```, if true : the join to debtor table is useful else we can skip it, that makes the request more faster.

Comment (jniles) [reply server/controllers/finance/patientInvoice.js:231]: Ah, I understand now.  LGTM.

**Review by jniles — APPROVED** [review]
bors r+

**Pull request merged**